diff --git a/CMakeLists.txt b/CMakeLists.txt
index dea25b1..875892b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -10,12 +10,19 @@ if(NOT CMAKE_BUILD_TYPE)
       FORCE)
 endif(NOT CMAKE_BUILD_TYPE)
 
+# for libraries and headers in the target directories
+set(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)
+set(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)
+
+# god-awful hack because it seems to not run correct tests to determine this:
+set(__CHAR_UNSIGNED___EXITCODE 1)
+
 set(CMAKE_CXX_STANDARD 20)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
 # note that if you haven't installed/built the toolkit with CoordGen, you'll
 # have problems with this.
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -g -rdynamic")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -g")
 if(CMAKE_BUILD_TYPE MATCHES "Release")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DNDEBUG -O2 -Wall -DRDK_BUILD_COORDGEN_SUPPORT=ON" )
 endif(CMAKE_BUILD_TYPE MATCHES "Release")
@@ -39,7 +46,7 @@ set(Boost_USE_STATIC_LIBS OFF)
 set(Boost_USE_MULTITHREADED OFF)
 set(Boost_USE_STATIC_RUNTIME OFF)
 find_package( RDKit COMPONENTS
-    ChemReactions Depictor
+    ChemReactions Depictor ChemTransforms
     MolDraw2D DistGeometry DistGeomHelpers
     MolAlign Optimizer ForceField ForceFieldHelpers
     Alignment Fingerprints MolTransforms EigenSolvers
@@ -78,8 +85,8 @@ include_directories(${CMAKE_SOURCE_DIR})
 
 add_subdirectory(rdchiral)
 
-include(CTest)
-add_subdirectory(test)
+# include(CTest)
+# add_subdirectory(test)
 
 # add the uninstall support
 configure_file(
diff --git a/cmake/FindRDKit.cmake b/cmake/FindRDKit.cmake
index 898b319..2caa907 100644
--- a/cmake/FindRDKit.cmake
+++ b/cmake/FindRDKit.cmake
@@ -32,7 +32,7 @@ set(RDKIT_FOUND "RDKit_FOUND")
 foreach(component ${RDKit_FIND_COMPONENTS})
   message( "Looking for RDKit component ${component}" )
   find_file( RDKit_LIBRARY_${component}
-    libRDKit${component}.so
+    libRDKit${component}${CMAKE_SHARED_LIBRARY_SUFFIX}
     PATH ${RDKIT_DIR}/lib NO_DEFAULT_PATH)
   message("RDKit_LIBRARY_${component} : ${RDKit_LIBRARY_${component}}")
   if(${RDKit_LIBRARY_${component}} MATCHES "-NOTFOUND$")
diff --git a/rdchiral/CMakeLists.txt b/rdchiral/CMakeLists.txt
index 56a3f19..9c8eb88 100644
--- a/rdchiral/CMakeLists.txt
+++ b/rdchiral/CMakeLists.txt
@@ -1,3 +1,5 @@
+add_link_options(-rdynamic)
+
 add_library(rdchiral SHARED
     rdkit_helper.cpp
     clean.cpp
